### PR TITLE
Allow disabling default failure predicates.

### PR DESCRIPTION
**Pull request by mergeconflict**
Works on #826

### CODE REVIEW

**Review by mum4k — CHANGES_REQUESTED** [review]
Looks like the CI failed, because README needs to be regenerated.

Could you try running this locally and uploading the commit?

```
tools/update_cli_readme_documentation.sh --mode fix
```

Additionally we should add both a unit test and an e2e test for the new feature. The unit test should live [here](https://github.com/envoyproxy/nighthawk/blob/main/test/options_test.cc) and the integration test probably [here](https://github.com/envoyproxy/nighthawk/blob/main/test/integration/test_integration_basics.py).

**Comment by mergeconflict** [comment]
> Could you try running this locally and uploading the commit?
> 
> ```
> tools/update_cli_readme_documentation.sh --mode fix
> ```

Done, thanks!

> Additionally we should add both a unit test and an e2e test for the new feature. The unit test should live [here](https://github.com/envoyproxy/nighthawk/blob/main/test/options_test.cc) and the integration test probably [here](https://github.com/envoyproxy/nighthawk/blob/main/test/integration/test_integration_basics.py).

I've added a unit test. I agree an e2e test would make sense, but I'm having some trouble constructing one, maybe you have some advice... My issue is that I can't find a way to configure NH test server to send error responses, which is necessary to verify that there aren't any failure predicates. Any suggestion?

**Comment by mum4k** [comment]
Without looking closer - could we achieve the same by pointing Nighthawk to a port the test server **isn't** listening on?

**Review by mum4k — CHANGES_REQUESTED** [review]
Looks great, glad this approach worked out. I have two minor requests aiming to make the integration test case more understandable.

File: test/integration/test_integration_basics.py [hunk 1]
```diff
@@ -647,6 +648,21 @@ def test_http_h1_failure_predicate(http_test_server_fixture):
   asserts.assertCounterEqual(counters, "benchmark.http_2xx", 1)
 
 
+def test_http_h1_no_default_failure_predicates(http_test_server_fixture):
+  """Test with no default failure predicates.
+
+  Point the client at a bogus port, but disable the default failure predicates.
+  Should result in successful execution despite pool connection failure.
+  """
+  root_uri = re.sub(":[0-9]+", ":123", http_test_server_fixture.getTestServerRootUri())
```
[the remark this answers is not in the document]
Comment (mergeconflict) [reply test/integration/test_integration_basics.py]: Done

File: test/integration/test_integration_basics.py [hunk 2]
```diff
@@ -647,6 +648,21 @@ def test_http_h1_failure_predicate(http_test_server_fixture):
...
+  parsed_json, _ = http_test_server_fixture.runNighthawkClient([
+      root_uri, "--duration", "5", "--rps", "500", "--connections", "1",
+      "--no-default-failure-predicates"
```
[the remark this answers is not in the document]
Comment (mergeconflict) [reply test/integration/test_integration_basics.py:661]: Done, I updated the docstring to be a bit more explicit.

**Review by mum4k — COMMENTED** [review]
Looks good, thank you for moving the code to the utility library.

Looks like the CI is failing due to formatting of Python files, could you execute `ci/do_ci.sh fix_format` and see if that helps?

**Review by mum4k — APPROVED** [review]
Thank you @mergeconflict !

**Pull request merged**